src/clojurewerkz/elastisch/native/conversion.clj

```diff
@@ -1448,3 +1450,70 @@
 (defn ^DeleteIndexTemplateRequest ->delete-index-template-request
   [template-name]
   (DeleteIndexTemplateRequest. template-name))
+
+;;
+;; Bulk Responses
+;;
+
+(defn bulk-item-response->map
+  [^BulkItemResponse item]
+  (let [res {:index (.getIndex item)
+             :_index (.getIndex item)
+             :type (.getType item)
+             :_type (.getType item)
+             :id (.getId item)
+             :_id (.getId item)
+             :version (.getVersion item)
+             :_version (.getVersion item)
+             :op-type (.getOpType item)
+             :failed? (.isFailed item)}]
+    (if (:failed? res)
+      (assoc res :failure-message (.getFailureMessage item))
+      res)))
+
+(defn bulk-response->map
+  [^BulkResponse response]
+  {:took (.getTookInMillis response)
+   :has-failures? (.hasFailures response)
+   :items (mapv bulk-item-response->map (.getItems response))})
+
+(defn remove-underscores [opts]
+  (reduce-kv (fn [m k v]
+               (assoc m (keyword (clojure.string/replace (name k) #"^_" "")) v))
+             {} opts))
+
+(defn get-bulk-item-action
+  [doc]
+  (cond (contains? doc "index") "index"
+        (contains? doc "delete") "delete"
+        :else nil))
+
+(defn ->action-requests
+  [a]
+  (loop [actions a
```

Comment:
I had a hard time implementing this with reduce because you need the next item if there is an index item. But feel free to improve this one with a reduce I would be curious how to handle this.